compiler/bin-js_of_ocaml/compile.ml

```diff
@@ -65,6 +74,52 @@ let output_gen ~standalone ~custom_header ~build_info ~source_map output_file f
   | `Stdout -> f stdout `Stdout
   | `Name name -> Filename.gen_file name (fun chan -> f chan `File)
 
+let find_source file =
+  match Builtins.find file with
+  | Some f -> Some (Source_map.Source_content.create (Builtins.File.content f))
+  | None ->
+      if Sys.file_exists file && not (Sys.is_directory file)
+      then
+        let content = Fs.read_file file in
+        Some (Source_map.Source_content.create content)
+      else None
+
+let build_source_map_with_offset
```

Comment:
It’s particularly obscure what this function does (apparently, turning a `Source_map.info` into a `Source_map.Standard.t` while ignoring the initial unmapped lines and using filenames from an existing `Source_map.Standard.t`…). A comment would save many minutes of code reading.

Reply:
I've done some refactoring and renaming. I did not add comment yet but it should be much easier to read.